File: .i3/status.rb
#!/usr/bin/env ruby
require 'json'

def calendar
  {
    full_text: Time.now.strftime('%a, %d of %b')
  }
end

def clock
  {
    full_text: Time.now.strftime('%I:%M:%S %p')
  }
end

puts({ version: 1 }.to_json)
puts '[[]'

loop do
  puts ',' << [calendar, clock].to_json
  STDOUT.flush
  sleep 1
end

File: .i3/status.js
#!/usr/bin/env node

'strict mode';

const exec = require('child_process').exec;
const moment = require('moment');

const calendar = function() {
    return  {
        full_text: moment().format('ddd, DD of MMM')
    };
};

const clock = function() {
    return {
        full_text: moment().format('hh:mm:ss A')
    };
};

console.log("{\"version\": 1}");

// Begin infinite array.
console.log("[");

(function main() {
    console.log(JSON.stringify([calendar(), clock()]) + ",");
    setTimeout(main, 1000);
})();
